### PR TITLE
Add the Wake Vision dataset

### PR DESCRIPTION
**Pull request by Ekhao**
# Add Dataset

* Dataset Name: Wake Vision
* Issue Reference: No reference

## Description

Wake Vision" is a large, high-quality dataset featuring over 6 million images, significantly exceeding the scale and diversity of current tinyML datasets (100x). This dataset includes images with annotations of whether each image contains a person. Additionally, it incorporates a comprehensive fine-grained benchmark to assess fairness and robustness, covering perceived gender, perceived age, subject distance, lighting conditions, and depictions.
The Wake Vision labels are derived from Open Image's annotations which are licensed by Google LLC under CC BY 4.0 license. The images are listed as having a CC BY 2.0 license. Note from Open Images: "while we tried to identify images that are licensed under a Creative Commons Attribution license, we make no representations or warranties regarding the license status of each image and you should verify the license for each image yourself.

## Checklist

* [x] Address all TODO's
* [x] Add alphabetized import to subdirectory's `__init__.py`
* [x] Run `download_and_prepare` successfully
* [x] Add [checksums file](https://www.tensorflow.org/datasets/add_dataset#2_run_download_and_prepare_locally)
* [x] Properly cite in `BibTeX` format
* [x] Add passing test(s)
* [x] Add test data
* [x] If using additional dependencies (e.g. `scipy`), use [lazy_imports](https://www.tensorflow.org/datasets/add_dataset#extra_dependencies) (if applicable)
* [x] Add data generation script (if applicable)
* [x] [Lint](https://www.tensorflow.org/datasets/add_dataset#5_check_your_code_style) code

### CODE REVIEW

**Comment by Ekhao** [comment]
Hello @fineguy,

Thanks for assigning this pull request to yourself. Do you have an estimate for how long it will take to review it?

File: tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
+"""wake_vision dataset."""
+
+import tensorflow_datasets.public_api as tfds
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py:3]: `import tensorflow_datasets as tfds`

File: tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import os
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py:4]: Fix imports:
https://www.tensorflow.org/datasets/contribute?_gl=1*9d64n*_up*MQ..*_ga*MTgwMjYxNjY4OS4xNzIzMDM5OTU3*_ga_W0YLR4190T*MTcyMzAzOTk1Ny4xLjAuMTcyMzAzOTk1Ny4wLjAuMA..#check_your_style_guide

File: tensorflow_datasets/image_classification/__init__.py [hunk 3]
```diff
@@ -96,3 +96,4 @@
 from tensorflow_datasets.image_classification.svhn import SvhnCropped
 from tensorflow_datasets.image_classification.uc_merced import UcMerced
 from tensorflow_datasets.image_classification.visual_domain_decathlon import VisualDomainDecathlon
+from tensorflow_datasets.image_classification.wake_vision import WakeVision
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/__init__.py:99]: This is a legacy folder, all new datasets should go into https://github.com/tensorflow/datasets/tree/master/tensorflow_datasets/datasets

File: tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                  tfds.features.ClassLabel(names=['No', 'Yes']),
+              'person_depiction':
+                  tfds.features.ClassLabel(names=['No', 'Yes']),
+              'non-person_depiction':
+                  tfds.features.ClassLabel(names=['No', 'Yes']),
+              'non-person_non-depiction':
+                  tfds.features.ClassLabel(names=['No', 'Yes']),
+          }
+      ),
+      # If there's a common (input, target) tuple from the
```
Comment (fineguy) [inline tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py:119]: Delete these comments.

**Review by fineguy — COMMENTED** [review]
Thanks for the PR, sorry that I took so long to get to it.

I left a few small comments, but overall looks good!

File: tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py (hunk 2 above)
Comment (Ekhao) [reply tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py]: I assume that you refer to this: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#313-imports-formatting

In that case I reordered the imports to first import standard libraries and afterwards tfds.

File: tensorflow_datasets/image_classification/__init__.py (hunk 3 above)
Comment (Ekhao) [reply tensorflow_datasets/image_classification/__init__.py]: I have moved the wake_vision folder to the new datasets folder and removed the above line. From what I understand, I do not need to register my dataset in any __init__.py file for the new folder.

I believe the confusion stems from this [page](https://www.tensorflow.org/datasets/add_dataset#add_an_import_for_registration) which says:

>  For example, if you're contributing to tensorflow/datasets, add the module import to its subdirectory's __init__.py (e.g. [image/__init__.py](https://github.com/tensorflow/datasets/tree/master/tensorflow_datasets/image/__init__.py).

Perhaps this should be updated.

File: tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py (hunk 4 above)
Comment (Ekhao) [reply tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py]: Done

File: tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py (hunk 1 above)
Comment (Ekhao) [reply tensorflow_datasets/image_classification/wake_vision/wake_vision_dataset_builder.py]: Done

**Pull request merged**